worker/src/core/loaders/message-worker/email.class.ts

```diff
@@ -66,21 +67,54 @@ class Email {
       .then((result) => map(result, 'get_messages_to_send_email'))
   }
 
-  sendMessage({
+  calculateHmac(campaignId: number, recipient: string) {
+    const version = config.get('unsubscribeHmac.version')
+    return crypto
+      .createHmac(
+        config.get(`unsubscribeHmac.${version}.algo`),
+        config.get(`unsubscribeHmac.${version}.key`)
+      )
+      .update(`${campaignId}.${recipient}`)
+      .digest('hex')
+  }
+
+  generateUnsubLink(campaignId: number, recipient: string): URL {
+    const version = config.get('unsubscribeHmac.version')
+    const hmac = this.calculateHmac(campaignId, recipient)
+    const link = new URL(`/unsubscribe/${version}`, config.get('frontendUrl'))
+    link.searchParams.append('c', campaignId.toString())
+    link.searchParams.append('r', recipient)
+    link.searchParams.append('h', hmac)
+    return link
+  }
+
+  appendUnsubToMessage(msg: string, unsubUrl: string): string {
+    return `${msg} <br> <hr> <p style="font-size:12px;color:#818284;line-height:2em">
```

Comment:
Petty would like to change the copywriting to the following:

[Postman.gov.sg](http://postman.gov.sg/) is a mass messaging platform used by the Singapore Government to communicate with stakeholders. For more information, please visit our [site](https://guide.postman.gov.sg/faqs/faq-recipients). 

If you wish to unsubscribe from similar emails from your sender, please click [here]() to unsubscribe and we will inform the respective agency.